pkg/operator/credentialsrequest/status.go

```diff
@@ -29,75 +26,41 @@ const (
+
+func (r *ReconcileCredentialsRequest) GetConditions(logger log.FieldLogger) ([]configv1.ClusterOperatorStatusCondition, error) {
 	_, credRequests, operatorIsDisabled, err := r.getOperatorState(logger)
 	if err != nil {
-		return fmt.Errorf("failed to get operator state: %v", err)
+		return []configv1.ClusterOperatorStatusCondition{}, fmt.Errorf("failed to get operator state: %v", err)
 	}
-
-	oldConditions := co.Status.Conditions
-	oldVersions := co.Status.Versions
-	oldRelatedObjects := co.Status.RelatedObjects
 	parentSecretExists, err := r.parentSecretExists()
 	if err != nil {
-		return errors.Wrap(err, "error checking if parent secret exists")
+		return []configv1.ClusterOperatorStatusCondition{}, errors.Wrap(err, "error checking if parent secret exists")
 	}
-	co.Status.Conditions = computeStatusConditions(oldConditions, credRequests, r.platformType, operatorIsDisabled,
-		parentSecretExists, r.Actuator.GetCredentialsRootSecretLocation(), logger)
-	co.Status.Versions = computeClusterOperatorVersions()
-	co.Status.RelatedObjects = buildExpectedRelatedObjects(credRequests)
-
-	// ClusterOperator should already exist (from the manifests payload), but recreate it if needed
-	if isNotFound {
-		if err := r.Client.Create(context.TODO(), co); err != nil {
-			return fmt.Errorf("failed to create clusteroperator %s: %v", co.Name, err)
-		}
-		logger.Info("created clusteroperator")
-	}
-
-	// Check if version changed, if so force a progressing last transition update:
-	if !reflect.DeepEqual(oldVersions, co.Status.Versions) {
-		logger.WithFields(log.Fields{
-			"old": oldVersions,
-			"new": co.Status.Versions,
-		}).Info("version has changed, updating progressing condition lastTransitionTime")
-		progressing := findClusterOperatorCondition(co.Status.Conditions, configv1.OperatorProgressing)
-		// We know this should be there.
-		progressing.LastTransitionTime = metav1.Time{Time: time.Now()}
-	}
-
-	// Update status fields if needed
-	if !clusteroperator.ConditionsEqual(oldConditions, co.Status.Conditions) ||
-		!reflect.DeepEqual(oldVersions, co.Status.Versions) ||
-		!reflect.DeepEqual(oldRelatedObjects, co.Status.RelatedObjects) {
+	return computeStatusConditions(credRequests, r.platformType, operatorIsDisabled, parentSecretExists, r.Actuator.GetCredentialsRootSecretLocation(), logger), nil
+}
 
-		err = r.Client.Status().Update(context.TODO(), co)
-		if err != nil {
-			return fmt.Errorf("failed to update clusteroperator %s: %v", co.Name, err)
-		}
-		logger.Debug("cluster operator status updated")
+func (r *ReconcileCredentialsRequest) GetRelatedObjects(logger log.FieldLogger) ([]configv1.ObjectReference, error) {
+	_, credRequests, _, err := r.getOperatorState(logger)
+	if err != nil {
+		return []configv1.ObjectReference{}, fmt.Errorf("failed to get operator state: %v", err)
 	}
+	return buildExpectedRelatedObjects(credRequests), nil
+}
 
-	return nil
+func (r *ReconcileCredentialsRequest) Name() string {
+	return "credreq"
```

Comment:
Why not just use the constant itself?

```suggestion
	return controllerName
```

Reply:
of course :facepalm: